### DISCUSSION TITLE
Apple Silicon (M3)+UTM & Hakchi works!

### DISCUSSION THREAD

**Discussion by NiGHTSaturn** [TeamShinkansen/Hakchi2-CE#466] [General]
Hi everyone,

After spending 3 days checking for the best way to have Hakchi working from an Apple Silicon computer, I decided to share how I did it from my M3 MBP.

For everything i've searched, these past days, I didn't find a single post on here or reddit on how to succeed.
_So I apologize if there's already some sort of guide or any other clues somewhere else._

I used UTM to emulate Windows 11. I wasn't successful with the arm64 version so I used x64.
The ISO can be downloaded using CrystalFetch, it's available on the Mac App Store.
I selected 8 Cores, forcing Multi Threading to add some speed and as much vRAM as possible (about 12GB in this case)
It's a lot slower than the ARM version, it's tedious using the internet (you have to refresh pages constantly, but the main purpose is Installing Hakchi) but it worked!

Make sure you are using the USB 3.0 Emulator in the UTM settings when the VM is off (Also sharing the USB devices with the host ON), if USB 2.0 is selected, you will have the Remote NDIS bug in the Device Manager.

Don't forget to install the Guest Tools and reboot!

Make sure the Firewall and any other protection is off.

I wanted to make sure the VM would ask me to confirm the USB connections (pop-up messages to confirm said device to connect).

Once you have Hakchi installed on the VM, go as normal, I connected the cable with a little USB-A to USB-C converter. 
Don't use a HUB! Try to make the connection as direct as possible.

Once FEL mode is activated, and Hakchi is waiting for the mini console to be connected, UTM will ask you if you want to connect the USB device to the VM, press confirm and when the kernel is starting to upload, release the reset button (in this case I had an SFC mini). Be on the lookout, UTM will ask once again if you want to connect another USB device "classic", confirm again so it can reboot properly.

I hope this will help someone. It IS possible to install Hakchi with an Apple Silicon Mac. It's just slow, but heh, if you don't have access to a PC and you HAVE to do it with your own Mac, here you go!

**Comment by raccoonman23** [comment]
I think it's not an ARM app; it's not easy to part non ARM apps to the Apple Silicon Macs. That's they don't bootcamp is becuase we can't buy Windows 11 on ARM liessen rigth now without hardware.

**Reply by NiGHTSaturn** [reply]
With UTM, you can select either processor emulation or selecting ARM for faster and native. What I did was emulating an x86/x64 PC with UTM. Hakchii is not ARM based indeed, which is why it was a pain to find a solution. It worked in the end.

**Comment by Silvergish** [comment]
Thank you! I've been looking for some way to get this to work on the new Macs. I used hackchi on an Intel Mac (using Windows 10) back in 2017. When I got a new Mac, I hoped that by now someone had created an ARM version or Mac version of hackchi. But no one seems to care enough to create one. 

Anyways, thank you for posting your solution. Hopefully this works for me too.

**Reply by NiGHTSaturn** [reply]
Hello! I sincerely hope it works out for you! Been about a year since I did it so I hope it still works. What I remember struggling the most was USB connection and keeping it there. Using the 3.0 Emulation in the settings should do the trick. Good luck!!

**Comment by lukets98** [comment]
THIS IS PRICELESS INFORMATION! THANK YOU!

**Reply by NiGHTSaturn** [reply]
My pleasure! :) I'm glad it still helps!
